src/inbox/mod_inbox.erl

```diff
@@ -559,16 +570,22 @@ get_inbox_unread(undefined, Acc, To) ->
     {ok, Count} = mod_inbox_backend:get_inbox_unread(User, Host, RemBareJIDBin),
     mongoose_acc:set(inbox, unread_count, Count, Acc).
 
-hooks(Host) ->
+hooks(HostType) ->
     [
-     {remove_user, Host, ?MODULE, remove_user, 50},
-     {remove_domain, Host, ?MODULE, remove_domain, 50},
-     {user_send_packet, Host, ?MODULE, user_send_packet, 70},
-     {filter_local_packet, Host, ?MODULE, filter_packet, 90},
-     {inbox_unread_count, Host, ?MODULE, inbox_unread_count, 80},
-     {get_personal_data, Host, ?MODULE, get_personal_data, 50}
+     {remove_user, HostType, ?MODULE, remove_user, 50},
+     {remove_domain, HostType, ?MODULE, remove_domain, 50},
+     {user_send_packet, HostType, ?MODULE, user_send_packet, 70},
+     {filter_local_packet, HostType, ?MODULE, filter_packet, 90},
+     {inbox_unread_count, HostType, ?MODULE, inbox_unread_count, 80},
+     {get_personal_data, HostType, ?MODULE, get_personal_data, 50}
     ].
 
+add_default_backend(Opts) ->
+    case lists:keyfind(backend, 2, Opts) of
```

Comment:
I think that key should be first, not second. It would be good to cover this in tests.